sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -508,30 +508,24 @@ class reduction_impl : private reduction_impl_base {
   }
 
   /// Constructs reduction_impl when the identity value is statically known.
-  // Note that aliasing constructor was used to initialize MRWAcc to avoid
-  // destruction of the object referenced by the parameter Acc.
   template <
       typename _T = T,
       enable_if_t<IsKnownIdentityOp<_T, BinaryOperation>::value> * = nullptr>
   reduction_impl(rw_accessor_type &Acc)
-      : MRWAcc(shared_ptr_class<rw_accessor_type>(
-            shared_ptr_class<rw_accessor_type>{}, &Acc)),
-        MIdentity(getIdentity()), InitializeToIdentity(false) {
+      : MRWAcc(new rw_accessor_type(Acc)), MIdentity(getIdentity()),
```

Comment:
will the "MRWacc" object take ownership and release the new memory?

Reply:
Yes, with the new changes MRWAcc will take ownership and will release memory

Reply:
OK, good